machine-learning/03-Training and AutoML.ipynb

```diff
@@ -1,45 +1,561 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "sdca rmse on trainset: 2.889902597526722, lgbm rmse on trainset: 117.9591141671402\r\n",
+            "sdca rmse on testset: 2.924784890718881, lgbm rmse on testset: 119.860254032606\r\n"
+          ]
+        }
+      ],
+      "source": [
+        "var sdcaModel = sdcaPipeline.Fit(trainTestSplit.TrainSet);\n",
+        "var lgbmModel = lgbmPipeline.Fit(trainTestSplit.TrainSet);\n",
+        "\n",
+        "// evaluate on train set\n",
+        "var sdcaTrainEval = sdcaModel.Transform(trainTestSplit.TrainSet);\n",
+        "var sdcaTrainMetric = context.Regression.Evaluate(sdcaTrainEval, \"y\");\n",
+        "\n",
+        "var lgbmTrainEval = lgbmModel.Transform(trainTestSplit.TrainSet);\n",
+        "var lgbmTrainMetric = context.Regression.Evaluate(lgbmTrainEval, \"y\");\n",
+        "\n",
+        "Console.WriteLine($\"sdca rmse on trainset: {sdcaTrainMetric.RootMeanSquaredError}, lgbm rmse on trainset: {lgbmTrainMetric.RootMeanSquaredError}\");\n",
+        "\n",
+        "// evaluate on test set\n",
+        "var sdcaTestEval = sdcaModel.Transform(trainTestSplit.TestSet);\n",
+        "var sdcaTestMetric = context.Regression.Evaluate(sdcaTestEval, \"y\");\n",
+        "\n",
+        "var lgbmTestEval = lgbmModel.Transform(trainTestSplit.TestSet);\n",
+        "var lgbmTestMetric = context.Regression.Evaluate(lgbmTestEval, \"y\");\n",
+        "Console.WriteLine($\"sdca rmse on testset: {sdcaTestMetric.RootMeanSquaredError}, lgbm rmse on testset: {lgbmTestMetric.RootMeanSquaredError}\");"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Example 2: Non-linear regression on LightGbm.\n",
+        "This example is to show the importance of hyper-parameter optimization. We will first create a non-linear dataset and two pipelines. One pipeline has `LightGbm` with `numberOfLeaves` set to 10, the other's set to 1000. Then train both pipelines with the same train dataset and Comparing their training performance by evaluating them on the same test dataset."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Create non-linear dataset\n",
+        "The code below artifacts non-linear dataset with a random residual, and loaded it as train/test `DataFrame`"
```

Comment:
```suggestion
        "The code below creates a non-linear dataset with a random residual. The dataset is loaded into train and test DataFrames"
```

Similar comment to above. I think we might want to change DataFrame to IDataView or remove it altogether.